Emlid 2.4.3 timetag

igor-gladkov opened the pull request.

staroselskii commented on src/rtkcmn.c:
```diff
@@ -1270,6 +1270,26 @@ extern int str2time(const char *s, int i, int n, gtime_t *t)
+    double ep[6];
+    char str[256],*p=str;
+    
+    if (i<0||(int)strlen(s)<i||(int)sizeof(str)-1<i) return -1;
```
Let's not stick to Tamoji's codestyle in our code! This is unreadable and unbearable :-(

staroselskii replied:
So in this particular line of code, I propose at least add spacing between operands and get rid of crazy casts to int as well.

staroselskii commented on src/rtkcmn.c:
```diff
@@ -1270,6 +1270,26 @@ extern int str2time(const char *s, int i, int n, gtime_t *t)
+    char str[256],*p=str;
+    
+    if (i<0||(int)strlen(s)<i||(int)sizeof(str)-1<i) return -1;
+    for (s+=i;*s&&--n>=0;) *p++=*s++; *p='\0';
```
It's a `memcpy`. It might be a good idea to use [strlcpy()](https://linux.die.net/man/3/strlcpy) or memcpy_s.

staroselskii commented on src/rtkcmn.c:
```diff
@@ -1270,6 +1270,26 @@ extern int str2time(const char *s, int i, int n, gtime_t *t)
+extern int str2time_(const char *s, int i, int n, gtime_t *t)
+{
+    double ep[6];
+    char str[256],*p=str;
```
Please, declare every variable of different type on a separate line.

staroselskii commented on src/stream.c:
```diff
@@ -574,14 +578,94 @@ static int statexserial(serial_t *serial, char *msg)
+{
+    char line[TIMETAG_LINE_MAXLEN];
+    
+    if ( fgets(line, TIMETAG_LINE_MAXLEN, fp_tag) == NULL ) return 0;
```
Let's move `return` statements to the next line from now on.

staroselskii reviewed (changes requested):
Could you please get rid of a nasty merge commit? A simple `git rebase` should do.

igor-gladkov committed:
stream.c: write and read timetags in text format

igor-gladkov committed:
stream.c: move return statements to the next line

igor-gladkov committed:
stream.c: change default file replay speed from 0.0 to 1.0

it allows to not specify the replay speed every time we run rtkrcv with
timetags
